File: LeetCode/Python/Easy/1700-Number-of-Students-Unable-to-Eat-Lunch/solution.py
from collections import deque

class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        sandwiches.reverse()
        
        stack=sandwiches

        queue=deque(students)

        while stack:
            students_ate=False

            q_size=len(queue)

            for iteration in range(0, q_size):
                student=queue.popleft()

                if student==stack[-1]:
                    students_ate=True

                    stack.pop()
                else:
                    queue.append(student)

            if not students_ate:
                break

        return len(queue)

File: LeetCode/Rust/Easy/1700-Number-of-Students-Unable-to-Eat-Lunch/solution.rs
use std::collections::{VecDeque};

impl Solution
{
    pub fn count_students(mut students: Vec<i32>, mut sandwiches: Vec<i32>) -> i32
    {
        let mut stack: Vec<i32>=sandwiches.clone();

        stack.reverse();

        let mut queue: VecDeque<i32>=students.clone().into_iter().collect::<VecDeque<i32>>();

        while !stack.is_empty()
        {
            let mut students_ate: bool=false;

            let q_size: usize=queue.len();

            for iteration in 0..q_size
            {
                if let Some(student)=queue.pop_front()
                {
                    if (*stack.last().unwrap())==student
                    {
                        students_ate=true;

                        stack.pop();
                    }
                    else
                    {
                        queue.push_back(student);
                    }
                }
            }

            if !students_ate
            {
                break;
            }
        }

        queue.len() as i32
    }
}